File: m01_String/Java/q003_isSequence.java
public class q003_isSequence {
    public boolean isSubsequence( String s , String t ){

        int i = 0 , j = 0;

        while( i < s.length() && j < t.length() ){
            if( s.charAt(i) == t.charAt(j) ){
                i++;
                j++;
            }else{
                j++;
            }
        }

        return i == s.length();
    }


    public static void main(String[] args) {
        String s = "node";
        String t = "neetcode";

        q003_isSequence q = new q003_isSequence();

        boolean check = q.isSubsequence( s , t );

        System.out.println("They are Subsequnce :" + check );

        return;
    }
}

File: m01_String/C++/q003_isSequence.cpp
#include<iostream>
using namespace std;

// ex : s = "node" , t = "neetcode"; true;
bool isSubsequence(string s, string t) {
    int i = 0 , j = 0;

    while( i < s.length() && j < t.length() ){
        if( s[i] == t[j] ){
            i++;
            j++;
        }else{
            j++;
        }
    }

    // after the loop the i become 4 as last check value to end it.
    return i == s.length();
}

int main(){
    string s = "node";
    string t = "neetcode";

    bool isSub = isSubsequence( s , t );

    cout<<"They  are Subsequene : "<<isSub;

    return 0;
}